icaruscode/Decode/fcl/CAEN_V1730_setup_icarus.fcl

```diff
@@ -109,10 +109,148 @@ icarus_V1730_setup_Run0: [
+}
+
+#
+# "Run 1": configuration good for data taken after March 2022.
+# 
+# This configuration is FROZEN: do not change it.
+#
+# Delays have been measured by Andrea Scarpelli, Animesh Chatterjee and Nick Suarez
+#   (see e.g. SBN DocDB 20283); the figure "43 ns" here is a rough guess from
+#   some of the measurements -- more precise ones may be available;
+#   the offset measured against a (supposedly) common reference via the
+#   Trigger Time Tag counter of the readout board settles to 48 ns
+#   (the counter tick is worth 16 ns).
+# The special channels are saved and their channel number encodes their meaning:
+# 0xT0CB:
+# * `T`: the content of the channel:
+#     * 0: PMT signal (routed into data product with empty instance name)
+#     * 1: trigger primitive signal (routed into "trgprim" data product instance name)
+#     * 2: RWM signal (routed into "RWM" data product instance name)
+#     * 3: GatedBES signal (routed into "BES" data product instance name)
+#     * 7: unknown/unspecified (routed into "unknwn" data product instance name)
+# * `C`: identifier of the crate (containing three boards), in three bits `cts`:
+#     * `c`: cryostat bit: `0` for east cryostat ("e"), `1` for west cryostat ("w")
+#     * `t`: TPC bit: `0` for east TPC ("e"), `1` for west TPC ("w")
+#     * `s`: side: `0` for bottom ("bot"), `1` for top ("top")
+#     For example, the digit 5 (0b101) is the crate `wetop`.
+# * `B`: board inside the crate: 1, 2 or 3.
+#
+icarus_V1730_West_setup_Run1: [
+
+  ### --------------------------------------------------------------------------
+  ###  WW
+  ### --------------------------------------------------------------------------
+  #
+  # top
+  #
+  { Name: "icaruspmtwwtop01"  TriggerDelay: " 0 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_trgprim_setup_Run1  Channel: 0x1071 } ] },
+  { Name: "icaruspmtwwtop02"  TriggerDelay: "43 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_BES_setup_Run1      Channel: 0x3072 } ] },
+  { Name: "icaruspmtwwtop03"  TriggerDelay: "86 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_misc_setup_Run1     Channel: 0x7073 } ] },
+  #
+  # bottom
+  #
+  { Name: "icaruspmtwwbot01"  TriggerDelay: " 0 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_trgprim_setup_Run1  Channel: 0x1061 } ] },
+  { Name: "icaruspmtwwbot02"  TriggerDelay: "43 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_misc_setup_Run1     Channel: 0x7062 } ] },
+  { Name: "icaruspmtwwbot03"  TriggerDelay: "86 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_misc_setup_Run1     Channel: 0x7063 } ] },
+  ### --------------------------------------------------------------------------
+  ### WE
+  ### --------------------------------------------------------------------------
+  #
+  # top
+  #
+  { Name: "icaruspmtwetop01"  TriggerDelay: " 0 ns"  SpecialChannels: [ { @table::icarus_V1730_16thChannel_trgprim_setup_Run1  Channel: 0x1051 } ] },
```

Comment:
I argue that since the signals are affected by those delays, they should be corrected natively here.
This allows to compare e.g. the BNB beam signal from the second readout board to the trigger primitive signal from the first readout board.